examples/image_segmentation/ios/ImageSegmenter/Services/Render.swift

```diff
@@ -0,0 +1,578 @@
+
+  var isPrepared = false
+
+  private(set) var outputFormatDescription: CMFormatDescription?
+
+  private var outputPixelBufferPool: CVPixelBufferPool?
+
+  private let metalDevice = MTLCreateSystemDefaultDevice()!
+
+  private var computePipelineState: MTLComputePipelineState?
+
+  private let image2 = UIImage(named: "bg1.jpeg")!
+
+  var inputTexture2: MTLTexture?
+
+  private var textureCache: CVMetalTextureCache!
+  let context: CIContext
+
+  let textureLoader: MTKTextureLoader
+
+  private lazy var commandQueue: MTLCommandQueue? = {
+    return self.metalDevice.makeCommandQueue()
+  }()
+
+  required init() {
+    let defaultLibrary = metalDevice.makeDefaultLibrary()!
+    let kernelFunction = defaultLibrary.makeFunction(name: "mergeColor")
+    do {
+      computePipelineState = try metalDevice.makeComputePipelineState(function: kernelFunction!)
+    } catch {
+      print("Could not create pipeline state: \(error)")
+    }
+    context =  CIContext(mtlDevice: metalDevice)
+    textureLoader = MTKTextureLoader(device: metalDevice)
+    inputTexture2 = try? textureLoader.newTexture(cgImage: image2.cgImage!)
+  }
+
+  func prepare(with imageSize: CGSize,
+               outputRetainedBufferCountHint: Int) {
+    reset()
+
+    (outputPixelBufferPool, _, outputFormatDescription) = allocateOutputBufferPool(with: imageSize,
+                                                                                   outputRetainedBufferCountHint: outputRetainedBufferCountHint)
+    if outputPixelBufferPool == nil {
+      return
+    }
+
+    var metalTextureCache: CVMetalTextureCache?
+    if CVMetalTextureCacheCreate(kCFAllocatorDefault, nil, metalDevice, nil, &metalTextureCache) != kCVReturnSuccess {
+      assertionFailure("Unable to allocate texture cache")
+    } else {
+      textureCache = metalTextureCache
+    }
+
+    isPrepared = true
+  }
+
+  func prepare(with formatDescription: CMFormatDescription,
+               outputRetainedBufferCountHint: Int,
+               needChangeWidthHeight: Bool = false) {
+    reset()
+
+    (outputPixelBufferPool, _, outputFormatDescription) = allocateOutputBufferPool(with: formatDescription,
+                                                                                   outputRetainedBufferCountHint: outputRetainedBufferCountHint, needChangeWidthHeight: needChangeWidthHeight)
+    if outputPixelBufferPool == nil {
+      return
+    }
+
+    var metalTextureCache: CVMetalTextureCache?
+    if CVMetalTextureCacheCreate(kCFAllocatorDefault, nil, metalDevice, nil, &metalTextureCache) != kCVReturnSuccess {
+      assertionFailure("Unable to allocate texture cache")
+    } else {
+      textureCache = metalTextureCache
+    }
+
+    isPrepared = true
+  }
+
+  func reset() {
+    outputPixelBufferPool = nil
+    outputFormatDescription = nil
+    textureCache = nil
+    isPrepared = false
+  }
+
+  func render(image: UIImage, segmentDatas: UnsafePointer<Float32>?) -> UIImage? {
```

Comment:
Why is only the mask at index 0 taken for display ?. This will work for selfie segmentation. But if you use a model like [deeplabv3 ](https://developers.google.com/mediapipe/solutions/vision/image_segmenter#deeplab-v3) which will provide segmentation masks for different categories in the different indices of the array, this won't work. 
The sample needs to support Deeply V3 too. The user should be allowed to switch between deeplabv3 and selfie segmenter model in the bottom bar.

Reply:
I will do it.